src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/brgemm_executor.cpp

```diff
@@ -0,0 +1,388 @@
+    brgCopyKernelConf.wei_n_blk = N_blk;
+    // B could come from strided tensor, must use copy_B_wei_stride if set.
+    brgCopyKernelConf.wei_tag = copy_B_wei_stride != 0 ? transpose ? dnnl_adbc : dnnl_acbd : transpose ? dnnl_ba : dnnl_ab;
+    brgCopyKernelConf.copy_B_wei_stride = copy_B_wei_stride;
+    // LDB here is for the target tensor, not source tensor
+    brgCopyKernelConf.LDB = LDB;
+    brgCopyKernelConf.N = N;
+    brgCopyKernelConf.N_tail = N_tail;
+    brgCopyKernelConf.N_blk = N_blk;
+    brgCopyKernelConf.K = K;
+    brgCopyKernelConf.K_blk = K;
+    brgCopyKernelConf.K_tail = 0;
+    brgCopyKernelConf.N_chunk_elems = brgCopyKernelConf.N_blk;
+    brgCopyKernelConf.b_dt_sz =
+        DnnlExtensionUtils::sizeOfDataType(static_cast<dnnl::memory::data_type>(brgCopyKernelConf.src_dt));
+    brgCopyKernelConf.tr_b_dt_sz =
+        DnnlExtensionUtils::sizeOfDataType(static_cast<dnnl::memory::data_type>(brgCopyKernelConf.src_dt));
+    brgCopyKernelConf.req_wei_vnni_downconvert = false;
+
+    if (is_with_amx) {
+        brgCopyKernelConf.isa = avx512_core_amx;
+        brgCopyKernelConf.s8s8_compensation_required = false;
+    } else {
+        brgCopyKernelConf.isa = dt_in0 == dnnl_data_type_t::dnnl_bf16 ? avx512_core_bf16 : avx512_core_vnni;
+    }
+
+    brgCopyKernelConf.has_zero_point_a = false;
+    brgCopyKernelConf.has_zero_point_b = false;
+    brgCopyKernelConf.src_zp_type = dnnl::impl::cpu::x64::none;
+
+#if defined(OPENVINO_ARCH_X86_64)
+    auto ret = create_brgemm_matmul_copy_b(brgCopyKernel, &brgCopyKernelConf);
+    if (ret != dnnl::impl::status_t::dnnl_success)
+        THROW_ERROR("cannot create_brgemm_matmul_copy_b kernel");
+#endif  // OPENVINO_ARCH_X86_64
+}
+
+void brgemmExecutor::copy_buffer_b(void* b, void* scratch_b) {
+    auto ptr_b = reinterpret_cast<uint8_t*>(b);
+    auto ptr_scartch_b = reinterpret_cast<uint8_t*>(scratch_b);
+    auto dataType = ov::element::bf16;
+    if (brgCopyBKernel) {
+        for (size_t nb = 0; nb < div_up(N, N_blk); nb++) {
+            auto N_stride = b_transposed ? ldb : 1;
+            auto pCopyKernel0In = ptr_b + nb * N_blk * dataType.size() * N_stride;
+            auto pCopyKernel0Out = ptr_scartch_b + nb * N_blk * brgVnniFactor * dataType.size();
+
+            auto ctx = jit_brgemm_matmul_copy_b_t::ctx_t();
+
+            const bool is_N_tail = (N - nb * N_blk < N_blk);
+            ctx.current_N_blk = is_N_tail ? N_tail : N_blk;
+            ctx.src = pCopyKernel0In;
+            ctx.tr_src = pCopyKernel0Out;
+            ctx.compensation_ptr = nullptr;
+            ctx.zp_a_compensation_ptr = nullptr;
+            ctx.zp_a_neg_value_ptr = nullptr;
+            ctx.current_K_start = 0;
+            ctx.current_K_iters = K;
+            (*brgCopyBKernel)(&ctx);
+        }
+    }
+}
+
+void brgemmExecutor::executeGemm(size_t m_blk, void* a, void* b, void* c, void* wsp, void* scratch_a, void* scratch_b) {
+    auto ptr_A = reinterpret_cast<uint8_t*>(a);
```

Comment:
The problem here is that maybe the B repack is only for int8 & bf16, if we want to later extend the executor then it may needs original B pointer.  Let me define this `executeGemm` as `executeGemmPackedB`